users/tests.py

```diff
@@ -137,10 +139,264 @@ def test_whoami(self):
+        auth = self.client.post(url, data=json.dumps(self._user), content_type='application/json')
+        fake_header = b'Bearer ' + json.loads(auth.content)['access'].encode()
+
+        data = {
+            'old_password': self._user['password'],
+            'password': 'SuperSecret',
+            'retype_password': 'SuperSecret1',
+        }
+        url = reverse('password-update', kwargs={'version': 'v1'})
+        response = self.client.post(url, data=json.dumps(data), content_type='application/json',
+                                    HTTP_AUTHORIZATION=fake_header)
+
+        self.assertEqual(response.content, b'{"password": ["Passwords mismatch"]}')
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_password_update_with_incorrect_password(self):
+        url = reverse('token-obtain-pair', kwargs={'version': 'v1'})
+        auth = self.client.post(url, data=json.dumps(self._user), content_type='application/json')
+        fake_header = b'Bearer ' + json.loads(auth.content)['access'].encode()
+
+        data = {
+            'old_password': self._user['password'],
+            'password': 'secret',
+            'retype_password': 'secret',
+        }
+        response_content = (b'{"password": ["This password is too short. It must contain at least '
+                            b'8 characters.", "This password is too common."]}')
+        url = reverse('password-update', kwargs={'version': 'v1'})
+        response = self.client.post(url, data=json.dumps(data), content_type='application/json',
+                                    HTTP_AUTHORIZATION=fake_header)
+
+        self.assertEqual(response.content, response_content)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_password_update_unauthorized(self):
+        url = reverse('password-update', kwargs={'version': 'v1'})
+        response = self.client.post(url, content_type='application/json')
+
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+class UserProfileUpdateTest(BaseSingleUserTest):
+    """Tests the profile_update endpoint. """
+
+    def test_user_profile_update(self):
+        url = reverse('token-obtain-pair', kwargs={'version': 'v1'})
+        auth = self.client.post(url, data=json.dumps(self._user), content_type='application/json')
+        fake_header = b'Bearer ' + json.loads(auth.content)['access'].encode()
+
+        data = {
+            'username': 'new_username',
+            'email': 'new_email@cusdeb.com',
+        }
+        url = reverse('user-profile-update', kwargs={'version': 'v1'})
+        response = self.client.post(url, data=json.dumps(data), content_type='application/json',
+                                    HTTP_AUTHORIZATION=fake_header)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_user_profile_update_without_required_data(self):
```

Comment:
Let's
* rename the method into `test_user_profile_update_with_missing_user_info`
* use the following docsting for the method:
   `"""Tests if it's not possible to update the current user profile with missing user info. """`